examples/python/pre-filter/pre-filter/handler.py

```diff
@@ -0,0 +1,260 @@
+
+
+def getManagedObjectTypeName(obj):
+    return mo.__class__.__name__.rpartition(".")[2]
+
+
+def getManagedObject(obj):
+    """
+    Convert an object as received from the event router in to a pyvmomi managed object
+    Args:
+        obj (object): object received from the event router
+    """
+    mo = None
+    try:
+        moref = obj['Value']
+        type = obj['Type']
+    except KeyError as err:
+        traceback.print_exc(limit=1, file=sys.stderr)  # providing traceback since it helps debug the exact key that failed
+        return 'Invalid JSON, required key not found > KeyError: {0}'.format(err), 400
+
+    if hasattr(vim, type):
+        typeClass = getattr(vim, type)
+        mo = typeClass(moref)
+        mo._stub = service_instance._stub
+        try:
+            debug(f'{bgc.OKBLUE}Managed object > {bgc.ENDC}{moref} has name {mo.name} and type {getManagedObjectTypeName(mo)}')
+            return mo
+        except vmodl.fault.ManagedObjectNotFound as err:
+            debug(f'{bgc.FAIL}{err.msg}{bgc.ENDC}')
+    return None
+
+
+def getViObjectPath(obj):
+    """
+    Gets the full path to the passed managed object
+    Args:
+        obj (vim.ManagedObject): VC managed object
+    """
+    path = ""
+    while obj != service_instance.content.rootFolder:
+        path = f'/{obj.name}{path}'
+        obj = obj.parent
+    return path
+
+
+def filterVcObject(obj, filter):
+    """
+    Takes a VC managed object and tests it against a filter
+    If it matches, a vRO input parameter object is returned
+    Args:
+        obj (vim.ManagedObject): VC managed Object
+        filter (str): Regex filter string
+    """
+
+    if obj and filter:
+        mo_type = getManagedObjectTypeName(obj)
+        debug(f'{bgc.OKBLUE}{mo_type} Filter > {bgc.ENDC}{filter}')
+        objPath = getViObjectPath(obj)
+        debug(f'{bgc.OKBLUE}{mo_type} Path > {bgc.ENDC}{objPath}')
+        if not re.search(filter, objPath):
+            debug(f'{bgc.WARNING}Filter "{filter}" does not match {mo_type} path "{objPath}". Exiting{bgc.ENDC}')
+            return f'Filter "{filter}" does not match {mo_type} path "{objPath}"', 200
+        else:
+            debug(f'{bgc.OKBLUE}Match > {bgc.ENDC}Filter matched {mo_type} path')
+
+
+def handle(req):
+    """
+    Handle a request to the function
+    Args:
+        req (str): request body
+    """
+
+    # Initialise a connection to vCenter
+    res = init()
+    if isinstance(res, tuple):  # Error state
+        return res
+    vcinfo = service_instance.RetrieveServiceContent().about
+    debug(f'Connected to {vcinfo.fullName} ({vcinfo.instanceUuid})')
+
+    # Load the Events that function gets from vCenter through the Event Router
+    debug(f'{bgc.HEADER}Reading Cloud Event: {bgc.ENDC}')
+    debug(f'{bgc.OKBLUE}Event > {bgc.ENDC}{req}')
+    try:
+        cevent = json.loads(req)
+    except json.JSONDecodeError as err:
+        return 'Invalid JSON > JSONDecodeError: {0}'.format(err), 400
```

Comment:
```suggestion
        return 'Invalid JSON > JSONDecodeError: {0}'.format(err), 500
```

Reply:
I don't know how/if this impacts on OpenFaaS handling of error states, but in this case a `JSONDecodeError` indicates a malformed request by the client rather than an internal error doesn't it?

Additionally, other python function examples return 400 for issues parsing the JSON request data.

Reply:
Sorry for causing confusion with the existing examples and how they treat the JSON (event) payload. Looks like I need to clean them up as well :)

Since the event router create the JSON (event) payload and I was considering the function(s) written for this project as one unit (router + function) I was suggesting 500 to indicate something is wrong with the internal logic of the router. A 400 could trip off users, looking for different reasons where the actual reason in 99% of the cases must be the router. 

If we treat the functions as "standalone", then 400 makes sense (as in the older examples) to indicate an error with the supplied payload (even though that will likely come from the router itself).